src/Grid/topology.jl

```diff
@@ -106,164 +149,125 @@ function _num_shared_vertices(cell_a::C1, cell_b::C2) where {C1, C2}
-        # Gather all cells which are connected via vertices
-        cell_neighbor_ids = Set{Int}()
-        for vertex ∈ vertices(cell)
-            for vertex_cell_id ∈ vertex_cell_table[vertex]
-                if vertex_cell_id != cell_id
-                    push!(cell_neighbor_ids, vertex_cell_id)
-                end
-            end
-        end
-        cell_neighbor_table[cell_id] = EntityNeighborhood(CellIndex.(collect(cell_neighbor_ids)))
-
-        # Any of the neighbors is now sorted in the respective categories
-        for cell_neighbor_id ∈ cell_neighbor_ids
-            # Buffer neighbor
-            cell_neighbor = cells[cell_neighbor_id]
-            # TODO handle mixed-dimensional case
-            getrefdim(cell_neighbor) == getrefdim(cell) || continue
-
-            num_shared_vertices = _num_shared_vertices(cell, cell_neighbor)
-
-            # Simplest case: Only one vertex is shared => Vertex neighbor
-            if num_shared_vertices == 1
-                for (lvi, vertex) ∈ enumerate(vertices(cell))
-                    for (lvi2, vertex_neighbor) ∈ enumerate(vertices(cell_neighbor))
-                        if vertex_neighbor == vertex
-                            push!(vertex_table[cell_id, lvi].neighbor_info, VertexIndex(cell_neighbor_id, lvi2))
-                            break
-                        end
-                    end
-                end
-            # Shared edge
-            elseif num_shared_vertices == 2
-                _add_single_edge_neighbor!(edge_table, cell, cell_id, cell_neighbor, cell_neighbor_id)
-            # Shared face
-            elseif num_shared_vertices >= 3
-                _add_single_face_neighbor!(face_table, cell, cell_id, cell_neighbor, cell_neighbor_id)
-            else
-                @error "Found connected elements without shared vertex... Mesh broken?"
-            end
-        end
-    end
-end
-
-function ExclusiveTopology(cells::Vector{C}) where C <: AbstractCell
-    # Setup the cell to vertex table
-    cell_vertices_table = vertices.(cells) #needs generic interface for <: AbstractCell
-    vertex_cell_table = Set{Int}[Set{Int}() for _ ∈ 1:maximum(maximum.(cell_vertices_table))]
-
-    # Setup vertex to cell connectivity by flipping the cell to vertex table
-    for (cellid, cell_vertices) in enumerate(cell_vertices_table)
-        for vertex in cell_vertices
-            push!(vertex_cell_table[vertex], cellid)
-        end
-    end
-
-    # Compute correct matrix size
-    celltype = eltype(cells)
-    max_vertices = 0
-    max_faces = 0
-    max_edges = 0
-    if isconcretetype(celltype)
-        max_vertices = nvertices(cells[1])
-        max_faces = nfaces(cells[1])
-        max_edges = nedges(cells[1])
+"Return the highest number of vertices, edges, and faces per cell"
+function _max_nentities_per_cell(cells::Vector{C}) where C
+    if isconcretetype(C)
+        cell = first(cells)
+        return nvertices(cell), nedges(cell), nfaces(cell)
```

Comment:
Since this is a corner case: It is expected that 1D elements dispatch on `nfaces` then?

Reply:
This is covered by `reference_faces(::Type{RefLine}) = ()` (which gets called by `faces` getting called by `nfaces`). So it is not required to explicitly dispatch `nfaces` for a new 1D element.